tsdb/head.go

```diff
@@ -1327,7 +1334,13 @@ func (h *Head) gc() {
 			ss = stringset{}
 			values[t.Name] = ss
 		}
-		ss.set(t.Value)
+
+		if _, ok := ss[t.Value]; !ok {
+			if t.Name != "" {
+				numLabelPairs++
+			}
+			ss.set(t.Value)
+		}
```

Comment:
Not sure, but I think `getOrCreateWithID()` should be sufficient for this.

Reply:
Only `getOrCreateWithID()` is not enough since it only accounts for addition of new label pairs and not deletion.